distributed/executor.py

```diff
@@ -1798,3 +1829,62 @@ def futures_of(o, executor=None):
             raise CancelledError(bad)
 
     return list(futures)
+
+
+@contextmanager
+def serialize_for_publish(x):
```

Comment:
That's effectively what I'm doing, but perhaps yours is more explicit and not likely to cause strange behaviour from exceptions.
I was planning on avoiding Delayed objects - if the user wants to publish those, they have to be submitted and the future used instead.